### PR TITLE
Cleaning up some CMake warnings in the dashboard

### PR DESCRIPTION
**Pull request by william76**
@trilinos/framework 

Cleaning up some CMake warning messages that are showing up on the dashboard in the Clean track by fixing some 'deprecated' features according to tribits messages by replacing:

- DEPLIBS with IMPORTEDLIBS
- DEFINES with TARGETDEFINES

Warning messges can be found here ([https://testing.sandia.gov/cdash/viewConfigure.php?buildid=2950039](https://testing.sandia.gov/cdash/viewConfigure.php?buildid=2950039))

Packages affected:
- ThreadPool
- Galeri
- Epetra

@bartlettroscoe please have a look to make sure that I didn't misinterpret the warning messages coming from tribits.

### CODE REVIEW

**Comment by mhoemmen** [comment]
Looks good to me, but @bartlettroscoe should have a look first.

File: packages/ThreadPool/test/hhpccg/CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ TRIBITS_ADD_EXECUTABLE(
   test_tpi_hhpccg
   COMM serial mpi
   SOURCES main.c CGSolver.c BoxPartitionIB.c tpi_vector.c dcrs_matrix.c
-  DEPLIBS pthread m
```
Comment (bartlettroscoe) [inline packages/ThreadPool/test/hhpccg/CMakeLists.txt]: Actually, `pthread` should be pulled in as the "Pthread" TPL defined in Trilinos/TPLsList.cmake and declared as an optional dependency in ThreadPool/cmake/Dependencies.cmake.   These tests should only get created if `${PACKAGE_NAME}_ENABLE_Pthread` is true.  You can address this by replacing `DEPSLIBS pthread m` with:

```
  TRIBITS_ADD_EXECUTABLE(
    ...
    EXCLUDE_IF_NOT_TRUE  ${PACKAGE_NAME}_ENABLE_Pthread
    IMPORTEDLIBS m
    ...
    )
```

As the documentation for `IMPORTEDLIBS` says:

* https://tribits.org/doc/TribitsDevelopersGuide.html#formal-arguments-tribits-add-executable

"So far, the only case where `IMPORTEDLIBS` has been shown to be necessary is to pass in the standard C math library `m`.  In every other case, a TriBITS TPL should be used instead."

Does that make sense?

File: packages/ThreadPool/test/hpccg/CMakeLists.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@ TRIBITS_ADD_EXECUTABLE(
   test_tpi_hpccg
   COMM serial mpi
   SOURCES main.c CGSolver.c BoxPartition.c tpi_vector.c
-  DEPLIBS pthread m
```
Comment (bartlettroscoe) [inline packages/ThreadPool/test/hpccg/CMakeLists.txt]: Same as above.  Handle `pthread` as a TPL.

File: packages/epetra/src/CMakeLists.txt [hunk 3]
```diff
@@ -245,5 +245,5 @@ TRIBITS_ADD_LIBRARY(
   epetra
   HEADERS ${HEADERS}
   SOURCES ${SOURCES}
-  DEFINES -DEPETRA_LIB_EXPORTS_MODE
+  TARGET_DEFINES -DEPETRA_LIB_EXPORTS_MODE
```
Comment (bartlettroscoe) [inline packages/epetra/src/CMakeLists.txt]: Yup, that is the right change as stated by the warning output.

**Comment by william76** [comment]
@bartlettroscoe I made the requested changes.  When you have an opportunity, do you mind having a quick look to verify that I made the modifications you were expecting?  If you're good with it, I'll go ahead and do a merge & squash the commits down to one commit.

**Comment by bartlettroscoe** [comment]
@william76:

> I made the requested changes. When you have an opportunity, do you mind having a quick look to verify that I made the modifications you were expecting? If you're good with it, I'll go ahead and do a merge & squash the commits down to one commit.

Did you see my review [above](https://github.com/trilinos/Trilinos/pull/1435#issuecomment-309796427)?

When you are ready to test and push, please use `./checkin-test-sems.sh --do-all --push` as per:

* https://github.com/trilinos/Trilinos/wiki/Policies-%7C-Safe-Checkin-Testing

Thanks,

-Ross

**Comment by william76** [comment]
There are some problems using TARGET_DEFINES with some of the epetra cmake files.  The rest of the changes worked fine though.  I am closing this PR and will put another in from a different branch on my fork which leaves out the CMakeLists.txt files in epetra that fail to configure when DEFINES is replaced with TARGET_DEFINES.

**Comment by bartlettroscoe** [comment]
> There are some problems using TARGET_DEFINES with some of the epetra cmake files. The rest of the changes worked fine though. I am closing this PR and will put another in from a different branch on my fork which leaves out the CMakeLists.txt files in epetra that fail to configure when DEFINES is replaced with TARGET_DEFINES.

What that means is that there were later targets that required these defines and got them accidentally due to the behavior of the deprecated `DEFINES` argument.  As the warning message stated, if you want all of the targets to get the same defines the just use the built-in CMake function `ADD_DEFINITIONS()`.  That is why I deprecated the `DEFINES` argument to avoid confusion like this!

**Pull request closed without merging**